### ISSUE TITLE
[Town Guards] Horde town guards are super twitchy.

### ISSUE THREAD

**Issue by Syfte-oss** [SunstriderEmu/BugTracker#1649]
I'm not even sure how to introduce this.  Aggroing town guards outside Shadowmoon Village in SMV and Thrallmar in HP is pure chaos.  This feels like it should be 3 or 4 separate reports.

- I was aggroing the guards from what seemed to be too far away despite being 10 levels above them.
- I would enter combat, then aggro them, then they'd reset seconds later.  If I didn't leave this would repeat over and over.
- Aggroing them seemed to cause one npc to mutli-spawn in place.  One of same npc would sometimes try to run up into the air to attack me but then reset before it reached me.
- The first time I pulled the Thrallmar gate guards four of them came after me which whittled down to 2 by the time I reach the Path of Glory.  The remaining two chased me almost to the Dark Portal when I had a choice between fighting them or facepulling a pack of demons.  I was not recording when this happened.
- When I tried to reproduce this while recording (3rd link below) I got knocked off my flying mount without anything appearing the combat log as the mounted patrol rocketed out of the wall and down the path.  I was then chain stunned and roflstomped before I could run away.

Chased to the Dark Portal:
https://i.imgur.com/PWDi3FH.jpg

Shadowmoon Village guards:
https://youtu.be/f3lHdlrJjnw

Thrallmar Guards:
https://youtu.be/4J1BeCXCzNU

**Comment by Jerrythemagicdog** [comment]
Thanks for the report!
> 1. I was aggroing the guards from what seemed to be too far away despite being 10 levels above them.

Keep in mind that guards have an increased aggro range and will definitely engage from further.

**Comment by Syfte-oss** [comment]
>guards have an increased aggro range

I believe I saw that mentioned in another bug report here.  It stood out to me in part because aggro radius in general was a common complaint on another private server.  I guess I underestimated how much you guys had tweaked it.
I hope you tweaking didn't deliberately include that marathon foot chase.  ;)

**Comment by Jerrythemagicdog** [comment]
No worries, it might be completely unreasonable but we always rely on testing it in an actual scenario and then making a judgement rather than sticking to something for better or worse, so thanks for the testing. 
About the guards spawning on top of other guards / civilians that's the guard posts at work there but the implementation is not polished yet - the guards will continue spawning until you deplete the guard post.

**Comment by Syfte-oss** [comment]
>the guards will continue spawning until you deplete the guard post.

And the mystery force that knocked me off my mount, was that also by design?

**Comment by Jerrythemagicdog** [comment]
That's an ability most guards in Outland have that dismount you, I think even creatures in Zul'Farrak have that (also on retail and Classic).

**Comment by Syfte-oss** [comment]
I have no memory of that but okay.  :)

One last question, are guards supposed to chase players that far.  I wandered too close to Allerian Stronghold while on a Belf and a guard chased me all the way back Stonebreaker Hold where two horde guards started smacking him but couldn't pull him off me.  It was like the alliance guard was ignoring threat.

**Comment by Jerrythemagicdog** [comment]
Probably something wrong on our end there, thanks!

**Comment by kelno** [comment]
Super thanks for the details and taking time to make videos

**Comment by Syfte-oss** [comment]
I'll record pulling the guards at those two towns again and we'll see what the feedback to that is.  As best I could tell the pulls were both unremarkable.  I got close enough to body pull, I ran away and they chased me until they were killed.  They didn't appear to be tethered to their patrol/spawn points and damage from another source (when I hadn't damaged them myself) didn't change their target.

**Comment by Syfte-oss** [comment]
I think we've made some progress.

I visted Allerian, Thrallmar and Stonebreaker and the normal guards on the ground were working properly.  It's the rider guards and the new, spawning guards that don't seem to be tethered.

First, my Allerian visit.  I have to get in the mounted guard's face to aggro him and he rests properly but then I aggro a mounted rider who chases almost to Stonebreaker when he breaks off as I'm crossing the river.  The only difference between this and my attempt 2 days ago is that this time it's a rider guard plus  I jumped twice as I crossed the river.  The rider guard resets at the beginning of the second jump.  Previously I had swum straight across.
Also I realized I was mistaken in an earlier post when I said that the Stonebreaker guards should have been able to pull him off of me.  The guards had a bucketful of healing aggro to overcome before that was going to happen.

Allerian first try:  https://youtu.be/40VZEa5Amkc

The second video is straight forward.  The Grunt spawns and follows me a ridiculous distance until an Honor Hold detachment kills it.
Thrallmar 2nd try:  https://youtu.be/OI-Ipw5Vlf4

And finally a visit to Stonebreaker.  Here I don't get any Grunts and the guards seem to reset normally.  If anything their reset distance feels too short.
Stonebreaker 3rd try:  https://youtu.be/z_jP8ZUgo1k
